resources/repo.rb

```diff
@@ -105,7 +105,12 @@ def repo = repository.repositoryManager.get(params.name)
 
   nexus3_api 'delete_repo' do
     action %i(create run)
-    content 'repository.repositoryManager.delete(args)'
+    content <<-EOS
+def repo = repository.repositoryManager.get(args)
+if (repo != null) {
+  repository.repositoryManager.delete(args)
```

Comment:
With this code your script will be idempotent; but not your resource.
You should follow the same logic but on the Chef side by adding a `not_if { current_resource.nil? }`